### PR TITLE
Remove Converting step and transform. Order inputs for ActionsDAG.

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/ActionsDAG.cpp [hunk 1]
```diff
@@ -531,6 +561,115 @@ bool ActionsDAG::empty() const
...
+                                    ErrorCodes::ILLEGAL_COLUMN);
+            }
+            else
+                throw Exception("Cannot convert column " + backQuoteIfNeed(res_elem.name) + " because "
+                                "it is non constant in source stream but must be constant in result",
+                                ErrorCodes::ILLEGAL_COLUMN);
+        }
+
+        if (!res_elem.type->equals(*src_node->result_type))
+        {
```
Comment (nikitamikhaylov) [inline src/Interpreters/ActionsDAG.cpp:637]: What happens in this scope? I don't understand. Do we simply cast result type to another type?

File: src/Interpreters/ActionsDAG.cpp [hunk 2]
```diff
@@ -531,6 +561,115 @@ bool ActionsDAG::empty() const
     return true;
 }
 
+ActionsDAGPtr ActionsDAG::makeConvertingActions(
+    const ColumnsWithTypeAndName & source,
+    const ColumnsWithTypeAndName & result,
```
Comment (nikitamikhaylov) [inline src/Interpreters/ActionsDAG.cpp:566]: What is result?

File: src/Interpreters/ExpressionActions.cpp [hunk 3]
```diff
@@ -163,11 +164,11 @@ void ExpressionActions::linearizeActions()
         {
             /// Argument for input is special. It contains the position from required columns.
             ExpressionActions::Argument argument;
-            argument.pos = required_columns.size();
+            // argument.pos = required_columns.size();
```
Comment (nikitamikhaylov) [inline src/Interpreters/ExpressionActions.cpp:167]: Remove?

**Review by nikitamikhaylov — APPROVED** [review]

File: src/Interpreters/ActionsDAG.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Interpreters/ActionsDAG.cpp:637]: Yes, we just simply apply CAST function
CAST has 2 args - column and new type name
So, we add constant with type name before. And then apply CAST)

File: src/Interpreters/ActionsDAG.cpp (hunk 2 above)
Comment (KochetovNicolai) [reply src/Interpreters/ActionsDAG.cpp:566]: Result is `ActionDAG` which converts block structure from source to result

Comment (KochetovNicolai) [reply src/Interpreters/ActionsDAG.cpp:566]: `result` is expected structure we should have after applying ActionsDAG

File: src/Interpreters/ExpressionActions.cpp (hunk 3 above)
Comment (KochetovNicolai) [reply src/Interpreters/ExpressionActions.cpp:167]: yes

**Pull request merged**